### PR TITLE
Saching/adding deep link api

### PR DESCRIPTION
**Pull request by sachingupta**
Changes include. 1. adding deepLink API 
2. refactoring few Interfaces to separate module file (will do more in separate CR)

### CODE REVIEW

File: src/MicrosoftTeams.interface.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+export interface ShowNotificationParameters {
+  message: string;
+  notificationType: NotificationTypes;
+}
+
+export interface ExecuteDeepLinkParameters {
+  deepLink: string;
+}
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.interface.ts]: Why only move these as opposed to all the enums/interfaces from MicrosoftTeams.ts?

File: src/MicrosoftTeams.ts [hunk 2]
```diff
@@ -861,6 +856,21 @@ export function showNotification(
...
 
+/**
+ * @private
+ * Hide from docs.
+ * ------
+ * execute deep link API.
+ * @param deepLink deep link.
+ */
+export function executeDeepLink(
+  deepLinkParameters: ExecuteDeepLinkParameters
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Do we really need an ExecuteDeepLinkParameters object? Couldn't we just have the developer pass the URL?

File: src/MicrosoftTeams.ts [hunk 3]
```diff
@@ -861,6 +856,21 @@ export function showNotification(
   sendMessageRequest(parentWindow, "showNotification", params);
 }
 
+/**
+ * @private
+ * Hide from docs.
+ * ------
+ * execute deep link API.
+ * @param deepLink deep link.
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: The comment doesn't match the actual parameters of the API.

File: src/MicrosoftTeams.ts [hunk 4]
```diff
@@ -861,6 +856,21 @@ export function showNotification(
   sendMessageRequest(parentWindow, "showNotification", params);
 }
 
+/**
+ * @private
+ * Hide from docs.
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Why mark this as @private? I think this is an API we actually want made available to all apps.

File: src/MicrosoftTeams.ts [hunk 5]
```diff
@@ -861,6 +856,21 @@ export function showNotification(
...
+ * ------
+ * execute deep link API.
+ * @param deepLink deep link.
+ */
+export function executeDeepLink(
+  deepLinkParameters: ExecuteDeepLinkParameters
+): void {
+  ensureInitialized(frameContexts.content);
+  const params = [deepLinkParameters.deepLink];
+  sendMessageRequest(parentWindow, "executeDeepLink", params);
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Seems like you are missing the error handling for this.

File: src/MicrosoftTeams.interface.ts (hunk 1 above)
Comment (sachingupta) [reply src/MicrosoftTeams.interface.ts]: Will move other in separate PR

File: src/MicrosoftTeams.ts (hunk 4 above)
Comment (sachingupta) [reply src/MicrosoftTeams.ts]: I am planning to keep it private till this is available in teams Public Ring

File: src/MicrosoftTeams.ts (hunk 2 above)
Comment (sachingupta) [reply src/MicrosoftTeams.ts]: will do that now.

File: src/MicrosoftTeams.ts [hunk 6]
```diff
@@ -861,6 +861,24 @@ export function showNotification(
...
+/**
+ * @private
+ * Hide from docs.
+ * ------
+ * execute deep link API.
+ * @param deepLink deep link.
+ */
+export function executeDeepLink(deepLink: string): void {
+  ensureInitialized(frameContexts.content);
+  const params = [deepLink];
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: NIT: You could just put this inline below rather than using an intermediate `params` variable.

**Pull request merged**